test_pull_requests

```diff
@@ -1106,6 +1239,24 @@ popd
 
     end
 
+    # The GitHub mergeable API is flaky, so we use an external file ``database'' to record
+    # pull request mergeability responses with a biased saturating counter so that pull
+    # request comment/label actions are only taken after a critical amount of net positive
+    # or negative API responses. However, a pull request is immediately chosen not to be 
+    # tested after the first unmergeable result, even if it takes longer for the UX comment
+    # and label to be applied.
+    #
+    # The saturation points are at 0 and 10, with an initial state of 0, so i.e. the default
+    # state is to consider the pull request mergeable, and this state will not change until
+    # a net of four unmergeable responses come in from the GitHub API. When set_mergeable is
+    # called, the counter for the specific pull ID is decremented if it hasn't saturated; 
+    # when set_not_mergeable is called, the counter is incremented unless it is saturated.
+    #
+    # TODO: if we saturate at 10, why do we threshold at 4?
```

Comment:
Are you talking about for the coreq case?  I think that's a bug.  Should be sharing a constant.